quickstart/backend/src/main/java/com/digitalasset/quickstart/service/AppInstallsApiImpl.java

```diff
@@ -47,142 +56,186 @@ public AppInstallsApiImpl(
         this.authenticatedPartyService = authenticatedPartyService;
     }
 
-    /**
-     * List all AppInstall contracts visible to the authenticated party.
-     */
     @Override
+    @WithSpan
     public CompletableFuture<ResponseEntity<List<org.openapitools.model.AppInstall>>> listAppInstalls() {
-        logger.info("Listing AppInstalls");
-        String party = authenticatedPartyService.getPartyOrFail();
-
-        return damlRepository.findActiveAppInstalls()
-                .thenApply(contracts -> {
-                    List<org.openapitools.model.AppInstall> result = contracts.stream()
-                            .filter(contract -> {
-                                // Filter to those relevant to the requesting party
-                                String dso = contract.payload.getDso.getParty;
-                                String provider = contract.payload.getProvider.getParty;
-                                String user = contract.payload.getUser.getParty;
-                                // The requesting party must be involved
-                                return party.equals(dso) || party.equals(provider) || party.equals(user);
-                            })
-                            .map(contract -> {
-                                org.openapitools.model.AppInstall model = new org.openapitools.model.AppInstall();
-                                model.setContractId(contract.contractId.getContractId);
-                                model.setDso(contract.payload.getDso.getParty);
-                                model.setProvider(contract.payload.getProvider.getParty);
-                                model.setUser(contract.payload.getUser.getParty);
-
-                                org.openapitools.model.Metadata metaModel = new org.openapitools.model.Metadata();
-                                metaModel.setData(contract.payload.getMeta.getValues);
-                                model.setMeta(metaModel);
-
-                                model.setNumLicensesCreated(contract.payload.getNumLicensesCreated.intValue());
-                                return model;
-                            })
-                            .collect(Collectors.toList());
-                    return ResponseEntity.ok(result);
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        methodSpan.addEvent("Starting listAppInstalls");
+        logger.info("listAppInstalls: retrieving AppInstalls for the requesting party");
```

Comment:
The fact that we don't have any useful information to include in the log is a good indication this should be at DEBUG and there should be a more informative log at INFO later in the method.